upload-api/stores/plans.js

```
@@ -22,7 +25,21 @@ export function usePlansStore(customerStore) {
     },
 
     set: async (customer, plan) => {
-      return await customerStore.updateProduct(customer, plan)
+      try {
+        const hasCustomerResponse = await billingProvider.hasCustomer(customer)
+        if (hasCustomerResponse.ok) {
+          await billingProvider.setPlan(customer, plan)
+          return await customerStore.updateProduct(customer, plan)
+        } else {
+          if (hasCustomerResponse.error) {
+            return hasCustomerResponse
+          } else {
+            return { error: new Failure(`billing provider does not have customer for ${customer}`) }
+          }
+        }
+      } catch (/** @type {any} */ err) {
+        return { error: err }
```

Just let it throw...this is what ucanto will do for us essentially.